src/seedu/addressbook/data/person/Address.java

```diff
@@ -3,56 +3,142 @@
-     */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = address;
-    }
-
-    /**
-     * Returns true if a given string is a valid person email.
-     */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
-    }
-
-    @Override
-    public String toString() {
-        return value;
-    }
-
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
-    }
-
-    @Override
-    public int hashCode() {
-        return value.hashCode();
-    }
-
-    public boolean isPrivate() {
-        return isPrivate;
-    }
+	public static final String EXAMPLE = "123, some street";
+	public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+	public static final String ADDRESS_VALIDATION_REGEX = ".+";
+
+	// public final String value;
+
+	private Block block;
+	private Street street;
+	private Unit unit;
+	private PostalCode postalCode;
+	private boolean isPrivate;
+
+	/**
+	 * Validates given address.
+	 *
+	 * @throws IllegalValueException
+	 *             if given address string is invalid.
+	 */
+	public Address(String address, boolean isPrivate) throws IllegalValueException {
+		this.isPrivate = isPrivate;
+		String[] splitAddress = address.split(", ");
+		if (!isValidAddress(splitAddress)) {
+			throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+		}
+		// this.value = address;
+		block = new Block(splitAddress[0]);
+		street = new Street(splitAddress[1]);
+		unit = new Unit(splitAddress[2]);
+		postalCode = new PostalCode(splitAddress[3]);
+
+	}
+
+	/**
+	 * Returns true if a given string is a valid person email.
+	 */
+	public static boolean isValidAddress(String[] splitAddress) {
+		return splitAddress.length==4;//return test.matches(ADDRESS_VALIDATION_REGEX);
+	}
+
+	@Override
+	public String toString() {
+		// return value;
+		return block.getBlock() + ", " + street.getStreet() + ", " + unit.getUnit() + ", " + postalCode.getPostalCode();
+	}
+
+	@Override
+	public boolean equals(Object other) {
+		/*
+		 * return other == this // short snstanceof Address // instanceof
+		 * handles nulls && this.value.equals(((Address) other).value)); //
+		 * state check
+		 */
+		return other == this
+				&& this.toString().equals(((Address) other).toString());
+	}
+
+	@Override
+	public int hashCode() {
+		return toString().hashCode();
+	}
+
+	public boolean isPrivate() {
+		return isPrivate;
+	}
+}
+
+class Block {
+
+	private String block;
+
+	public Block() {
+		block = "";
+	}
```

Comment:
No need to create above constructor, since Java will automatically assign the variable `block` to empty string.